firefox-ios/Client/Frontend/Browser/DownloadToast.swift

```diff
@@ -49,11 +49,25 @@ class DownloadToast: Toast {
 
     var downloads: [Download] = []
 
-    var percent: CGFloat = 0.0 {
+    // Returns true if one or more downloads have encoded data (indicated via response `Content-Encoding` header).
+    // If at least one download has encoded data, we cannot get a correct total estimate for all the downloads.
+    // In that case, we do not show descriptive text. This will be improved in a later rework of the download manager.
+    // FXIOS-9039
+    var hasContentEncoding: Bool {
+        return downloads.contains(where: { $0.hasContentEncoding ?? false })
+    }
+
+    var percent: CGFloat? = 0.0 {
         didSet {
             UIView.animate(withDuration: 0.05) {
                 self.descriptionLabel.text = self.descriptionText
-                self.progressWidthConstraint?.constant = self.toastView.frame.width * self.percent
+
+                if let percent = self.percent {
+                    self.progressWidthConstraint?.constant = self.toastView.frame.width * percent
+                } else {
+                    self.progressWidthConstraint?.constant = 0
+                }
```

Comment:
Is this effectively hiding it by giving it a zero width? I think that's probably fine, however the control would still be visible to VoiceOver (I believe) and there could be other side effects of having it visible in the view hierarchy. Would probably be better to actually hide the view, but not a big deal and doesn't need to block this PR.

Reply:
Oh, that's a good point! Nice catch. Let me look into this. Adding `isHidden` should be pretty straightforward in any case too.

Reply:
Ok, looks like VoiceOver (tested via Accessibility Inspector) doesn't read out anything to do with the bar width. But I still swapped to hiding it instead of setting the size to 0 since it's straightforward anyway.

Reply:
I think it's generally a good policy to actually hide views rather than using a 0 width but it's definitely NAB, doesn't need to hold up this PR. Thanks again for digging into this 👍

Reply:
I'll keep in mind in the future! 👍  I've already made the change so all good. :) Just waiting for Bitrise to give me the signal all's good.